### ISSUE TITLE
Search requires the user to hit enter to trigger

### ISSUE THREAD

**Issue by jackson-laskey** [unbounded-enterprise/sample-app#42]
Not the worst thing, but would be great to automatically trigger a search after a user has stopped inputting into the field.

Currently in order to trigger a search, they must hit the enter key after filling out the search field.

Pre Enter:

![Image](https://user-images.githubusercontent.com/25757801/231835994-e1c5d202-329f-4334-bfae-9c7b3269faaf.png)

Post Enter:

![Image](https://user-images.githubusercontent.com/25757801/231836044-94496bfe-d8a3-4ab4-8e8b-b74a0de58d14.png)

**Comment by jackson-laskey** [comment]
@dstraffon 

Part of the justification here is that entering a search will trigger an API call. So the tradeoff is the feeling of responsiveness for the user vs sparing costs to the app. Same tradeoff has now come up for the page change on the collection explorer page, although that one is responsive and doesn't require an enter.

I am down to make the search responsive, but let me know first if you think that this is still a good idea

**Comment by dstraffon** [comment]
Makes sense. I do like the responsiveness for collection search but think it's less necessary when it comes to searching by serial number. Would be fine saving on api calls when it comes to the collection page and requiring a trigger to execute the search.

Thinking if we keep it this way it would probably be good to add a search button in addition to the "enter" key to trigger search, just so users aren't expecting instant results.

![Image](https://user-images.githubusercontent.com/25757801/234109424-87293702-fe81-43b9-bb9d-19e3f50e6501.png)

Lmk what you think, could seeing an argument for having them both function the same way

**Comment by jackson-laskey** [comment]
Think search button should go right next to the search bar or have the
sorts in between?

**Comment by dstraffon** [comment]
Assuming the sort function also requires an enter to trigger then I think I prefer the search button all the way to the right so it's the last component in the sequence if that makes sense. Sharing two alternate versions for comparison though



![Image](https://user-images.githubusercontent.com/25757801/234295710-f37a7df2-d2ea-4348-823c-547ac1551ffe.png)

![Image](https://user-images.githubusercontent.com/25757801/234295862-72a0e237-7154-4d44-b547-02cbfa82b043.png)
